activerecord/lib/active_record/persistence.rb

```diff
@@ -403,6 +409,12 @@ def _delete_record(constraints) # :nodoc:
         connection.delete(dm, "#{self} Destroy")
       end
 
+      def _insert_attributes(attributes)
+        raise ArgumentError, "Empty list of attributes passed" if attributes.blank?
+
+        return attributes unless record_timestamps
```

Comment:
Persistence shouldn't know about timestamps, that's why this method if here. Also no need for the return statement.